nipype/algorithms/modelgen.py

```diff
@@ -476,7 +476,7 @@ def _concatenate_info(self, infolist):
                                  sum(nscans[0:(i + 1)])
                         infoout.onsets[j].extend(onsets.tolist())
                 for j, val in enumerate(info.durations):
-                    if len(val) > 1:
+                    if len(val) > 0:
```

Comment:
i think this can simply be: `if len(val):`

also could you please add your test to test_modelgen? it should fail without this change, and pass with it? there is already a section that checks concatenation.